File: server/src/models/DecisionLog.js
/**
 * Pseudocode:
 * 1) Document DecisionLog shape.
 * 2) Store human decisions about balancing changes.
 */

/**
 * @typedef {Object} DecisionLog
 * @property {string} decisionId
 * @property {"easy"|"balanced"|"hard"} configId
 * @property {number=} stageId
 * @property {Object<string, unknown>} change
 * @property {string} rationale
 * @property {string[]} evidenceLinks
 * @property {string} timestamp
 */

export const DECISION_LOG_DOCS = {};

File: server/src/models/DecisionLog.java
package models;

import java.util.List;
import java.util.Map;

/**
 * Pseudocode:
 * 1) Document DecisionLog shape.
 * 2) Store human decisions about balancing changes.
 */
public class DecisionLog {
  public String decisionId;
  public String configId;
  public Integer stageId;
  public Map<String, Object> change;
  public String rationale;
  public List<String> evidenceLinks;
  public String timestamp;
}
